translate/convert/convert.py

```diff
@@ -229,7 +229,20 @@ def run(self, argv=None):
     def processfile(
         self, fileprocessor, options, fullinputpath, fulloutputpath, fulltemplatepath
     ):
-        if options.timestamp and _output_is_newer(fullinputpath, fulloutputpath):
+        # The `fulloutputpath`s parsed when `onefile` is enabled don't exist, which causes
+        # the `_output_is_newer` check to return False. This means that even if there is
+        # no content update, the POT/PO files are regenerated. When `onefile` is enabled,
+        # This check sets the output  path used by `_output_is_newer` to the output file
+        # provided to the command, which will exist. It remains `fulloutputpath` otherwise.
+        try:
+            if options.multifilestyle == "onefile":
+                timecheck_output_path = options.output
+            else:
+                timecheck_output_path = fulloutputpath
+        except AttributeError:
+            timecheck_output_path = fulloutputpath
```

Comment:
It's unclear from the code where you expect the attribute to be missing. I think this would make it easier to read:

```suggestion
        if getattr(options, "multifilestyle", None) == "onefile":
            timecheck_output_path = options.output
        else:
            timecheck_output_path = fulloutputpath
```